pkg/kademlia/routing.go

```diff
@@ -131,7 +131,7 @@ func (rt *RoutingTable) GetBuckets() (k []dht.Bucket, err error) {
 func (rt *RoutingTable) FindNear(id dht.NodeID, limit int) ([]*proto.Node, error) {
 	//if id is in the routing table
 	n, err := rt.nodeBucketDB.Get(id.Bytes())
-	if n != nil {
+	if n != nil && (rt.self.GetId() != id.String()) { // for bootstrapping, we don't want to stop when we find ourself
```

Comment:
this is a surprising check - sometimes the routing table should return yourself.

i'd prefer fixing this at the higher level, see comment above.

Reply:
perhaps more importantly, FindNear should not short circuit if the node is in the routing table. perhaps out of scope for this pr, but FindNear should always return the `limit` closest nodes to id, whether or not id is actually in the routing table or not

Reply:
I told @jenlij completely wrong then. 

I thought this method should return the node we are looking for if present otherwise return the k closest.

If there's a case we should return our self from our routing table, I'll adjust. I just couldn't think of a use case and this seemed much more simple.